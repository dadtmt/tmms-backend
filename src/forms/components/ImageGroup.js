import React, { PropTypes } from 'react'
import { ControlLabel, FormControl, FormGroup, Image } from 'react-bootstrap'

const ImageGroup = ({ input, label }) => (
  <FormGroup>
    <ControlLabel>{label}</ControlLabel>
    <FormControl {...input} type='text' placeholder='Image url' />
    <Image src={input.value} rounded />
  </FormGroup>
)

ImageGroup.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired
}

export default ImageGroup
