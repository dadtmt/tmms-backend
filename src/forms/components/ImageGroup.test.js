import React from 'react'
import renderer from 'react-test-renderer'

import ImageGroup from './ImageGroup'

it('should render correctly', () => {
  const props = {
    input: {
      value: 'http://image/png'
    },
    label: 'Image url'
  }
  expect(
    renderer.create(<ImageGroup {...props} />).toJSON()
  ).toMatchSnapshot()
})
