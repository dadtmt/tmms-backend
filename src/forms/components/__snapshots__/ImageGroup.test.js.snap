exports[`test should render correctly 1`] = `
<div
  className="form-group">
  <label
    className="control-label"
    htmlFor={undefined}>
    Image url
  </label>
  <input
    className="form-control"
    id={undefined}
    placeholder="Image url"
    type="text"
    value="http://image/png" />
  <img
    className="img-rounded"
    src="http://image/png" />
</div>
`;
